src/workerd/jsg/jsvalue.h

```diff
@@ -425,16 +428,27 @@ inline kj::Maybe<T&> JsValue::tryGetExternal(Lock& js, const JsValue& value) {
   return kj::Maybe<T&>(*static_cast<T*>(value.inner.As<v8::External>()->Value()));
 }
 
-template <typename T>
+template <Codepoint T>
 inline kj::Array<T> JsString::toArray(Lock& js, WriteOptions options) const {
+  v8::String::ValueView view(js.v8Isolate, inner);
   if constexpr (kj::isSameType<T, kj::byte>()) {
-    KJ_ASSERT(inner->ContainsOnlyOneByte());
     auto buf = kj::heapArray<kj::byte>(inner->Length());
-    inner->WriteOneByte(js.v8Isolate, buf.begin(), 0, buf.size(), options);
```

Comment:
Do we have reason to assume that WriteOneByte is less efficient than creating a string view and then copying the data? Both of them will be performing flattening (if needed) and then copy the data, so if the V8 implementation is reasonably sane there should be little difference.
In terms of correctness, the memcpy approach disregards the options field, for example for String::NO_NULL_TERMINATION, so this will result in different behavior.